lib/ansible/modules/cloud/google/gcp_compute_address.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2017 Google
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# ----------------------------------------------------------------------------
+#
+#     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+#
+# ----------------------------------------------------------------------------
+#
+#     This file is automatically generated by Magic Modules and manual
+#     changes will be clobbered when the file is regenerated.
+#
+#     Please read more about how to change this file at
+#     https://www.github.com/GoogleCloudPlatform/magic-modules
+#
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+################################################################################
+# Documentation
+################################################################################
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ["preview"],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: gcp_compute_address
+description:
+    - Represents an Address resource.
+      Each virtual machine instance has an ephemeral internal IP address and,
+      optionally, an external IP address. To communicate between instances on
+      the same network, you can use an instance's internal IP address. To
+      communicate with the Internet and instances outside of the same network,
+      you must specify the instance's external IP address.
+      Internal IP addresses are ephemeral and only belong to an instance for
+      the lifetime of the instance; if the instance is deleted and recreated,
+      the instance is assigned a new internal IP address, either by Compute
+      Engine or by you. External IP addresses can be either ephemeral or
+      static.
+short_description: Creates a GCP Address
+version_added: 2.6
+author: Google Inc. (@googlecloudplatform)
+requirements:
+    - python >= 2.6
+    - requests >= 2.18.4
+    - google-auth >= 1.3.0
+options:
+    state:
+        description:
+            - Whether the given object should exist in GCP
+        required: true
+        choices: ['present', 'absent']
+        default: 'present'
+    address:
+        description:
+            - The static external IP address represented by this
+              resource. Only IPv4 is supported.
+        required: false
+    description:
+        description:
+            - An optional description of this resource.
+        required: false
+    name:
+        description:
+            - Name of the resource.
```

Comment:
It'd be nice to include any restrictions on the name (can it have spaces? Dashes? Umlaüts?)